File: content/control_flow/defer-finally.java
void write(String fileName, String text) {
	File file = new File(fileName);
	PrintWriter writer;
	try {
		file.createNewFile();
		writer = new PrintWriter(file)
		writer.println(text);
	} catch (IOException e) {
		throw new RuntimeException(e);
	} finally {
		writer.close(); // finally always executes at the end
	}
}

File: content/control_flow/defer-finally.go
func write(fileName string, text string) {
	file, err := os.Create(fileName)
	if err != nil {
		panic(err)
	}
	defer file.Close() // if WriteString fails, it'll make sure that file is closed
	_, err = io.WriteString(file, text)
	if err != nil {
		panic(err)
	}
}
